### PR TITLE
Added support for sticky bit and execution bit to both be set on perm…

### LINKED ISSUE #30159: [storage] DataLake Path Permissions Support for Sticky Bit while Execute is enabled
### Library name

Azure.Storage.Files.DataLake

### Please describe the feature.

Following the update to the Storage Datalake REST API documentation for [Path Update Permissions](https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/update) regarding the x-ms-permissions

image

We will look to update the functionality of PathPermissions
https://docs.microsoft.com/en-us/dotnet/api/azure.storage.files.datalake.models.pathpermissions?view=azure-dotnet

To follow this documentation, as well as the parsers to convert from string to PathPermissions.

### PR DESCRIPTION
**Pull request by amnguye**
Added support for sticky bit and execution bit to both be set on permissions on a path

Resolves https://github.com/Azure/azure-sdk-for-net/issues/30159

### CODE REVIEW

**Comment by seanmcc-msft** [comment]
@tg-msft, this PR involves a public interface change.  Could you take a look?

**Review by seanmcc-msft — APPROVED** [review]

**Comment by seanmcc-msft** [comment]
Please wait until Azure SDK approved before merging.

File: sdk/storage/Azure.Storage.Files.DataLake/api/Azure.Storage.Files.DataLake.netstandard2.0.cs [hunk 1]
```diff
@@ -933,6 +933,7 @@ public static partial class PathAccessControlExtensions
         public static string ToAccessControlListString(System.Collections.Generic.IList<Azure.Storage.Files.DataLake.Models.PathAccessControlItem> accessControlList) { throw null; }
         public static string ToOctalRolePermissions(this Azure.Storage.Files.DataLake.Models.RolePermissions rolePermissions) { throw null; }
         public static string ToSymbolicRolePermissions(this Azure.Storage.Files.DataLake.Models.RolePermissions rolePermissions) { throw null; }
+        public static string ToSymbolicRolePermissions(this Azure.Storage.Files.DataLake.Models.RolePermissions rolePermissions, bool stickyBit = false) { throw null; }
```
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Files.DataLake/api/Azure.Storage.Files.DataLake.netstandard2.0.cs:936]: Can we drop the `= false` here?  Ideally `= false` would be what you get when you use the other existing overload.  It's pretty confusing for most customers which overload gets invoked when they pass `ToSymbolicRolePermissions(permissions)` and trawling through https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/expressions#1164-overload-resolution is no fun.

**Review by tg-msft — APPROVED** [review]
There's one small nit I think we should change, but otherwise this looks fine.

File: sdk/storage/Azure.Storage.Files.DataLake/api/Azure.Storage.Files.DataLake.netstandard2.0.cs (hunk 1 above)
Comment (amnguye) [reply sdk/storage/Azure.Storage.Files.DataLake/api/Azure.Storage.Files.DataLake.netstandard2.0.cs]: Done

**Pull request merged**